src/api/handlers/reports.go

```diff
@@ -0,0 +1,44 @@
+package handlers
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/lungria/spendshelf-backend/src/report"
+	"go.uber.org/zap"
+)
+
+type ReportsHandler struct {
+	generator report.Generator
+	logger    *zap.SugaredLogger
+}
+
+type GetQuery struct {
+	Start time.Time `form:"start" time_format:"unix"`
+	End   time.Time `form:"end" time_format:"unix"`
+}
+
+type GetResponse struct {
+	Report []report.Element `json:"report"`
+}
+
+func NewReportsHandler(generator report.Generator, logger *zap.SugaredLogger) *ReportsHandler {
+	return &ReportsHandler{generator: generator, logger: logger}
+}
+
+// HandleGet return all existing categories
+func (handler *ReportsHandler) HandleGet(c *gin.Context) {
+	var query GetQuery
+	err := c.BindQuery(&query)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, ResponseFromError(err))
+		return
+	}
+	reportResponse, err := handler.generator.GetReport(c, query.Start, query.End)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, ResponseFromError(err))
+		return
+	}
```

Comment:
What will the behavior of the report handler if nothing found on a specified period?

Reply:
api returned entity with field equals to null before, after https://gith ub.com/lungria/spendshelf-backend/pull/13/commits/bd2097f20684c56bebb2ec365af2ba29c573f8a2#diff-76434d2157e2ce5df37272308bd721daR68 
returns entity with empty array field |

```
{
    "report": []
}
```

Reply:
Ok